### PR TITLE
LC - Fix rendering bugs and improve parser/render logic

### PR DESCRIPTION
**Pull request by leandro**
## Context

This PR fix bugs related to overall documents parsing and rendering, where some conditional blocks were somehow being ignored and not being parsed properly.

This is related to this [clubhouse task](https://app.clubhouse.io/lawmatics/story/2211/docx-condition-issues).

[ch2211]

### CODE REVIEW

File: lib/lm_docstache/conditional_block.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    def self.inline_blocks_from_paragraph(paragraph)
+      node_set = Nokogiri::XML::NodeSet.new(paragraph.document, [paragraph])
+      conditional_blocks = []
+      scanner = StringScanner.new(paragraph.text)
+      matches = []
+
+      while scanner.scan_until(BLOCK_MATCHER)
+        next if matches.include?(scanner.matched)
+
+        content = scanner.captures[4]
```
Comment (daniloisr) [inline lib/lm_docstache/conditional_block.rb:62]: not a big deal, but would be nice to have a comment with an example of the `scanner.captures` content :)

**Review by daniloisr — APPROVED** [review]
LGTM, nice code.
One optional note: Would be nice to have a test to cover the problem reported on [[ch2211]](https://app.clubhouse.io/lawmatics/story/2211/docx-condition-issues)

File: lib/lm_docstache/conditional_block.rb (hunk 1 above)
Comment (frederico) [reply lib/lm_docstache/conditional_block.rb:62]: I agree with @daniloisr that's not very clear what the results are.

File: lib/lm_docstache/parser.rb [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        text = text_node.text
+
+        next unless text =~ VARIABLE_MATCHER
+
+        text.gsub!(VARIABLE_MATCHER) { |_match| data[$1].to_s }
+        text_node.content = text
+      end
+    end
+
+    def condition_from_match_data(match)
```
Comment (frederico) [inline lib/lm_docstache/parser.rb:151]: Over here is very welcome to also have examples.

**Review by frederico — APPROVED** [review]
LGTM. Just needs a lot more testing to make sure we're not breaking other stuff.

File: spec/integration_spec.rb [hunk 3]
```diff
@@ -67,10 +67,6 @@ module TestData
       expect(document.usable_tags.count).to be(30)
     end
 
-    it 'has the expected amount of role tags' do
-      expect(document.role_tags.count).to be(6)
```
Comment (rchasman) [inline spec/integration_spec.rb:71]: Why remove this test?

File: lib/lm_docstache/block.rb [hunk 4]
```diff
@@ -1,71 +0,0 @@
...
-      else
-        if @data.get(@name).is_a? Array
-          :loop
-        else
-          :conditional
-        end
-      end
-    end
-
-    def loop?
```
Comment (rchasman) [inline lib/lm_docstache/block.rb:27]: talked about this, but we can't just remove this functionality without recreating it elsewhere :p

**Review by rchasman — CHANGES_REQUESTED** [review]

File: spec/integration_spec.rb (hunk 3 above)
Comment (leandro) [reply spec/integration_spec.rb:71]: Because I removed the method from the `Document` class.

File: lib/lm_docstache/block.rb (hunk 4 above)
Comment (leandro) [reply lib/lm_docstache/block.rb:27]: As we spoke on private in Slack, we'll drop support for it now, and come up with a new solution for that in upcoming versions.

File: lib/lm_docstache/parser.rb (hunk 2 above)
Comment (leandro) [reply lib/lm_docstache/parser.rb:151]: Done in 640e537.

File: lib/lm_docstache/conditional_block.rb (hunk 1 above)
Comment (leandro) [reply lib/lm_docstache/conditional_block.rb:62]: Done in 640e537.

**Review by rchasman — APPROVED** [review]
We will be putting back the loop functionality within the next couple weeks most likely. Great refactor 🎉

**Pull request merged**